### ISSUE TITLE
Feature Request : Automatic Max, output in Watts

### ISSUE THREAD

**Issue by BenjiKimba** [lumapu/ahoy#1632]
## Hardware
  * [ ] ESP8266
  * [ ] ESP32
  * [ ] Raspberry Pi

Modelname: ______
Retailer URL: ______

### nRF24L01+ Module
* [ ] nRF24L01+ you verified this is a **Plus model** capable of the required 256kBit/s mode
* [ ] **square dot** indicates original Nordic Semicon chip 
* [ ] **round dot** indicates copy-cat / counterfeit SI labs chip

### Antenna:
* [ ] circuit board
* [ ] external antenna (SMA)

### Power Stabilization:
* [ ]  100uF Electrolytic Capacitor 
connected between +3.3V and GND (Pin 1 & 2) of the NRF Module
* [ ] Voltage stabilizing motherboard

### Version / Git SHA: 
Version: _._.__
Github Hash: _______

### Build & Flash Method:
* [ ] AhoyDTU Webinstaller
* [ ] VSCode - Platform IO
* [ ] Arduino
* [ ] ESP Tools

### Debugging:
* [ ] USB Serial Log (attached)
* [ ] Setup settings (use our templates ... to be added)

I have a HMS 1600 t4 with 2 solar panels on it delivering 800W max.
Now I want to add a third panel for more light sensitivity but do not want more output dan 800W.
Is it possible to modify the Active Power Control to do this automatically at his own ?
Its not the same as zero export, no need for data outside, only watching at AC Power and if the value go above 800W (threshold )
than pull the output lower.
Thanks in advance

**Comment by rmayergfx** [comment]
You don´t need any automation for this. HMS-1600 can be set from 600 to max with AhoyDTU, so if you set the limit in "W" it will always reduce the output power to the desired value, regardless of what is currently present. If you are below the output power, nothing happens at first; only when the value is above it is it reduced. This is done internally by the inverter; this automatic function is built in at the factory.

2 panels together with a HMS-1600 was not the best solution. You should always go as close as possible to the upper limit of the inverter with the individual modules, so that you can always get the maximum out of low light incidence or shading. If I have modules that only have a maximum output of 400W each, the yield is reduced when there is less sunlight; if I have modules that have 500W or more, the yield is higher even on bad days.
If a module is to be retrofitted anyway, then perhaps also check where it is mounted, as this can bring additional yield, see north-south orientation

**Comment by BenjiKimba** [comment]
Yes, I forgot to tell that the third panel is not in an optimal place, only for a few hours, so if I set the output
manually at 800W ( 200W per panel I Think ?), then I have at max. 600W, min 400W ?
I have 505W panels on it and want to add an 425W, they are cheap those days

**Comment by rmayergfx** [comment]
No, you didn´t understood, read the manual for your inverter. you are reducing the output to a max. of 800W. If this is not reached, it will not reduce anything.

**Comment by BenjiKimba** [comment]
I think only if you have all 4 panels on it, you can not choose the panels individual.
I have no place for 4 panels.

**Comment by knickohr** [comment]
If you want max power 800W

- you have to set the inverter to 800W for 4 panels
-  to 1067W for 3 panels
- to 1600W for 2 panels /unlimited)

**Comment by BenjiKimba** [comment]
That's true, but the 3th panel is not in optimal position, also one of the 2 panels became shading in Winter,
with my suggestion you can optimize the Yield dynamically  for this.
Not higher as 800W, but as Hight as possible.

**Comment by knickohr** [comment]
This is not possible until now.

A future version of Ahoy will have an „demand optimized power control“.

Then you can set the SetPoint to -800W and your panels and inverter is able to deliver max. 800W to the grid.

But it’s still a beta version  running on my DTU.

For more information, please join the Discord https://discord.com/channels/984173303147155506/1211365440253726851

**Comment by BenjiKimba** [comment]
Thank you for your effort on this.
I have only some Arduino experience and no idea about MqTT and all that stuff,
but I think its also with that possible, the same way how to make a zero export, but then 
reading the value of AC Power from the ESP ?

**Comment by BenjiKimba** [comment]
@knickohr 
No need for an extra button, only improved Active Power Control.
AhoyDTU knows  the capacity of the inverter and the panels add to it written in Inverter/ Max Module Power.
So he can decide to compensate the panels automatically even if you want to make a zero export .

By the way, no problem for me if someone want replied to me in German language.
